Fix a bug where garbage blocks sometimes don't fall down properly.

Patch by Dmitrii Tisnek

http://savannah.nongnu.org/bugs/?func=detailitem&item_id=10988
http://bugs.gentoo.org/show_bug.cgi?id=88747

--- crack-attack-1.1.12-r1/src/Grid.cxx
+++ crack-attack-1.1.12-r1/src/Grid.cxx
@@ -312,8 +312,8 @@
   // keep track of the bounds on the shattered area
   if (y + garbage.height > shatter_top)
     shatter_top = y + garbage.height;
-  if (y < shatter_bottom)
-    shatter_bottom = y;
+  if (y - garbage.height < shatter_bottom)
+    shatter_bottom = y - garbage.height;
 
   shatter_count += garbage.width * garbage.height;
 
